tests/blackbox_test.go

```diff
@@ -0,0 +1,205 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package blackbox
+
+import (
+	"fmt"
+	"net/http"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/coreos/ignition/tests/types"
+)
+
+func (server *HTTPServer) Config(w http.ResponseWriter, r *http.Request) {
+	w.Write([]byte(`{
+	"ignition": { "version": "2.0.0" },
+	"storage": {
+		"files": [{
+		  "filesystem": "root",
+		  "path": "/foo/bar",
+		  "contents": { "source": "data:,example%20file%0A" }
+		}]
+	}
+}`))
+}
+
+func (server *HTTPServer) Contents(w http.ResponseWriter, r *http.Request) {
+	w.Write([]byte(`asdf
+fdsa`))
+}
+
+type HTTPServer struct{}
+
+func (server *HTTPServer) Start() {
+	http.HandleFunc("/contents", server.Contents)
+	http.HandleFunc("/config", server.Config)
+
+	s := &http.Server{Addr: ":8080"}
+	go s.ListenAndServe()
+}
+
+func TestMain(m *testing.M) {
+	server := &HTTPServer{}
+	server.Start()
+	os.Exit(m.Run())
+}
+
+func TestIgnitionBlackBox(t *testing.T) {
+	tests := createTests()
+
+	for _, test := range tests {
+		t.Run(test.Name, func(t *testing.T) {
+			outer(t, test, false)
+		})
+	}
+}
+
+func TestIgnitionBlackBoxNegative(t *testing.T) {
+	tests := createNegativeTests()
+
+	for _, test := range tests {
+		t.Run(test.Name, func(t *testing.T) {
+			outer(t, test, true)
+		})
+	}
+}
+
+func PreCleanup(t *testing.T) {
+	mountpoints, _ := exec.Command(
+		"findmnt", "-l", "-o", "target").CombinedOutput()
+	points := strings.Split(string(mountpoints), "\n")
+	for i := len(points) - 1; i >= 0; i-- {
+		for _, pat := range []string{"/tmp/hd1p*", "/tmp/ignition-files*"} {
+			match, err := filepath.Match(pat, points[i])
+			if err != nil {
+				t.Log(err)
+			}
+			if match {
+				_, _ = exec.Command("umount", points[i]).CombinedOutput()
+			}
+		}
+	}
+}
+
+func outer(t *testing.T, test types.Test, negativeTests bool) {
+	PreCleanup(t)
+	t.Log(test.Name)
+
+	path := os.Getenv("PATH")
+	cwd, _ := os.Getwd()
+	_ = os.Setenv("PATH", fmt.Sprintf(
+		"%s:%s", filepath.Join(cwd, "bin/amd64"), path))
+
+	var rootLocation string
+
+	// Setup
+	for i, disk := range test.In {
+		// There may be more partitions created by Ignition, so look at the
+		// expected output instead of the input to determine image size
+		imageSize := calculateImageSize(test.Out[i].Partitions)
+
+		// Finish data setup
+		for _, part := range disk.Partitions {
+			if part.GUID == "" {
+				part.GUID = generateUUID(t)
+			}
+			updateTypeGUID(t, part)
+		}
+		setOffsets(disk.Partitions)
+		for _, part := range test.Out[i].Partitions {
+			updateTypeGUID(t, part)
+		}
+		setOffsets(test.Out[i].Partitions)
+
+		// NOTE: the image file is written into cwd because sgdisk fails when
+		// the file is located in /tmp/
```

Comment:
Moved `TMPDIR` to `/var/tmp`, removed this comment, and updated `disk.ImageFile` to be inside of `TMPDIR`.